module/planning/FailureManagement/src/FailureManagement.cpp

```diff
@@ -0,0 +1,154 @@
+
+#include "extension/Behaviour.hpp"
+#include "extension/Configuration.hpp"
+#include "extension/behaviour/Script.hpp"
+
+#include "message/behaviour/state/Stability.hpp"
+#include "message/input/Sensors.hpp"
+#include "message/motion/Limbs.hpp"
+#include "message/planning/Recover.hpp"
+
+#include "utility/support/yaml_expression.hpp"
+
+namespace module::planning {
+
+    using extension::Configuration;
+    using extension::behaviour::Script;
+    using extension::behaviour::ScriptRequest;
+    using message::behaviour::state::Stability;
+    using message::input::Sensors;
+    using message::motion::BodySequence;
+    using message::planning::Recover;
+    using utility::support::Expression;
+
+    enum class State { STABLE, UNSTABLE, FALLING };
+
+    double smooth(double value, double new_value, double alpha) {
+        return alpha * value + (1.0 - alpha) * new_value;
+    }
+
+    FailureManagement::FailureManagement(std::unique_ptr<NUClear::Environment> environment)
+        : BehaviourReactor(std::move(environment)) {
+
+        on<Configuration>("FailureManagement.yaml").then([this](const Configuration& config) {
+            // Use configuration here from file FailureManagement.yaml
+            this->log_level = config["log_level"].as<NUClear::LogLevel>();
+
+            cfg.falling.gyro.mag.mean       = config["falling"]["gyro"]["mag"]["mean"].as<Expression>();
+            cfg.falling.gyro.mag.unstable   = config["falling"]["gyro"]["mag"]["unstable"].as<Expression>();
+            cfg.falling.gyro.mag.falling    = config["falling"]["gyro"]["mag"]["falling"].as<Expression>();
+            cfg.falling.gyro.mag.smoothing  = config["falling"]["gyro"]["mag"]["smoothing"].as<Expression>();
+            cfg.falling.acc.mag.mean        = config["falling"]["acc"]["mag"]["mean"].as<Expression>();
+            cfg.falling.acc.mag.unstable    = config["falling"]["acc"]["mag"]["unstable"].as<Expression>();
+            cfg.falling.acc.mag.falling     = config["falling"]["acc"]["mag"]["falling"].as<Expression>();
+            cfg.falling.acc.mag.smoothing   = config["falling"]["acc"]["mag"]["smoothing"].as<Expression>();
+            cfg.falling.acc.angle.mean      = config["falling"]["acc"]["angle"]["mean"].as<Expression>();
+            cfg.falling.acc.angle.unstable  = config["falling"]["acc"]["angle"]["unstable"].as<Expression>();
+            cfg.falling.acc.angle.falling   = config["falling"]["acc"]["angle"]["falling"].as<Expression>();
+            cfg.falling.acc.angle.smoothing = config["falling"]["acc"]["angle"]["smoothing"].as<Expression>();
+
+            cfg.getup.angle         = config["getup"]["angle"].as<Expression>();
+            cfg.getup.gyro_recovery = config["getup"]["gyro_recovery"].as<Expression>();
+            cfg.getup.acc_recovery  = config["getup"]["acc_recovery"].as<Expression>();
+        });
+
+        on<Provide<Recover>, Trigger<Sensors>>().then([this](const RunInfo& info, const Sensors& sensors) {
```

Comment:
should this not be root level?

Reply:
`Recover` indicates something has gone wrong, but this is more just watching for something to go wrong.